document/core/syntax/instructions.rst

```diff
@@ -583,19 +585,21 @@ Instructions in this group are concerned with linear :ref:`memory <syntax-mem>`.
 
 Memory is accessed with |LOAD| and |STORE| instructions for the different :ref:`number types <syntax-numtype>`.
 They all take a *memory immediate* |memarg| that contains an address *offset* and the expected *alignment* (expressed as the exponent of a power of 2).
+The type of the *offset* corresponds to the *index type* of the memory which will
+be either |U32| (for 32-bit memories) or |U64| (for 64-bit memories).
```

Comment:
I don't think that's correct. It will be decoded/parsed as u64 in all cases (because parser/decoder don't know the types), but there is a range check in the validator.